File: Python/37-数字在排序数组中出现的次数.py
# -*- coding:utf-8 -*-
class Solution:
    def GetNumberOfK(self, data, k):
        # write code here
        # find leftmost index
        def biSearch(data, key):
            low, high = 0, len(data) - 1
            while low <= high:
                mid = (low + high) // 2
                if data[mid] < key:
                    low = mid + 1
                else:
                    high = mid - 1
            return low
        return biSearch(data, k + 0.5) - biSearch(data, k - 0.5)

File: C++/37-数字在排序数组中出现的次数.cpp
class Solution
{
public:
    int GetNumberOfK(vector<int> data, int k)
    {
        return biSearch(data, k + 0.5) - biSearch(data, k - 0.5);
    }
    int biSearch(vector<int> data, double key)
    {
        int low = 0, high = data.size() - 1;
        while (low <= high)
        {
            int mid = (low + high) / 2;
            if (data[mid] < key)
                low = mid + 1;
            else
                high = mid - 1;
        }
        return low;
    }
};